src/display_utils.rs

```diff
@@ -71,12 +68,17 @@ impl Display for SpaceOrNewline {
 
 /// A value that displays a comma-separated list of values.
 /// When pretty-printed (using {:#}), it displays each value on a new line.
-pub struct DisplayCommaSeparated<'a, T: fmt::Display>(&'a [T]);
+pub(crate) struct DisplayCommaSeparated<I, T: fmt::Display>(I)
+where
+    I: IntoIterator<Item = T> + Clone;
 
-impl<T: fmt::Display> fmt::Display for DisplayCommaSeparated<'_, T> {
+impl<I, T: fmt::Display> fmt::Display for DisplayCommaSeparated<I, T>
+where
+    I: IntoIterator<Item = T> + Clone,
+{
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         let mut first = true;
-        for t in self.0 {
+        for t in self.0.clone() {
```

Comment:
was the clone needed here, we seem to only call `fmt` on `t`?

Reply:
Yes, we are never actually cloning anything expensive here, but we need to add the `Clone` generic bound in order to be able to use the `IntoIterator` trait.

This `.clone()` concretely always clones
 - either just a reference
 - or an iterator object (but not the underlying structure being iterated on)

Reply:
@iffyio , I can roll it back to the non-generic version, and just implement comma separation manually in the Values Display impl, if you want.

Reply:
Yeah I think it might be better to make it explicit, with the new impl seems like it wouldn't be obvious if we were to go on to later clone something expensive accidentally

Reply:
@iffyio : I reverted the generic `indented_list` implementation and reworked the Values Display implementation not to use it.